File: common/src/main/java/dev/chasem/cobblemonextras/commands/EmptyBox.java
package dev.chasem.cobblemonextras.commands;

import com.cobblemon.mod.common.Cobblemon;
import com.cobblemon.mod.common.api.storage.NoPokemonStoreException;
import com.cobblemon.mod.common.api.storage.pc.PCBox;
import com.cobblemon.mod.common.api.storage.pc.PCPosition;
import com.cobblemon.mod.common.api.storage.pc.PCStore;
import com.cobblemon.mod.common.api.storage.pc.link.PCLink;
import com.cobblemon.mod.common.api.storage.pc.link.PCLinkManager;
import com.cobblemon.mod.common.net.messages.client.storage.pc.OpenPCPacket;
import com.mojang.brigadier.CommandDispatcher;
import com.mojang.brigadier.arguments.IntegerArgumentType;
import com.mojang.brigadier.context.CommandContext;
import dev.chasem.cobblemonextras.CobblemonExtras;
import dev.chasem.cobblemonextras.permissions.CobblemonExtrasPermissions;
import net.minecraft.server.command.ServerCommandSource;
import net.minecraft.server.network.ServerPlayerEntity;
import net.minecraft.text.Text;

import static net.minecraft.server.command.CommandManager.argument;
import static net.minecraft.server.command.CommandManager.literal;

public class EmptyBox {

    public void register(CommandDispatcher<ServerCommandSource> dispatcher) {
        dispatcher.register(
                literal("emptybox")
                        .requires(src -> CobblemonExtrasPermissions.checkPermission(src, CobblemonExtras.permissions.EMPTYBOX_PERMISSION))
                        .then(argument("box", IntegerArgumentType.integer(1, Cobblemon.config.getDefaultBoxCount()))
                        .executes(this::execute))
        );
    }

    private int execute(CommandContext<ServerCommandSource> ctx) {
        if (ctx.getSource().getPlayer() != null) {
            ServerPlayerEntity player = ctx.getSource().getPlayer();
            PCStore playerPc = null;
            try {
                playerPc = Cobblemon.INSTANCE.getStorage().getPC(player.getUuid());
            } catch (NoPokemonStoreException e) {
                player.sendMessage(Text.of("Error accessing PC..."));
                return -1;
            }
            Integer boxNum = ctx.getArgument("box", Integer.class) - 1;
            PCBox box = playerPc.getBoxes().get(boxNum);
            if (box == null) {
                player.sendMessage(Text.of("Error accessing box... " + boxNum));
                return -1;
            }
            for (int i = 0; i < 30; i++) {
                playerPc.remove(new PCPosition(boxNum, i));
            }
            player.sendMessage(Text.of("Box " + (boxNum + 1) + " has been emptied."));
        } else {
            ctx.getSource().sendError(Text.of("Sorry, this is only for players."));
        }
        return 1;
    }
}

File: common/src/main/kotlin/dev/chasem/cobblemonextras/commands/EmptyBox.kt
package dev.chasem.cobblemonextras.commands

import com.cobblemon.mod.common.Cobblemon
import com.cobblemon.mod.common.api.storage.NoPokemonStoreException
import com.cobblemon.mod.common.api.storage.pc.PCPosition
import com.cobblemon.mod.common.api.storage.pc.PCStore
import com.cobblemon.mod.common.config.CobblemonConfig
import com.mojang.brigadier.CommandDispatcher
import com.mojang.brigadier.arguments.IntegerArgumentType
import com.mojang.brigadier.context.CommandContext
import dev.chasem.cobblemonextras.CobblemonExtras
import dev.chasem.cobblemonextras.permissions.CobblemonExtrasPermissions
import net.minecraft.commands.CommandSourceStack
import net.minecraft.commands.Commands
import net.minecraft.network.chat.Component
import net.minecraft.server.level.ServerPlayer

class EmptyBox {
    fun register(dispatcher: CommandDispatcher<CommandSourceStack>) {
        dispatcher.register(
                Commands.literal("emptybox")
                        .requires { src: CommandSourceStack? -> CobblemonExtrasPermissions.checkPermission(src, CobblemonExtras.permissions.EMPTYBOX_PERMISSION) }
                        .then(Commands.argument<Int>("box", IntegerArgumentType.integer(1, Cobblemon.config.defaultBoxCount))
                                .executes { ctx: CommandContext<CommandSourceStack> -> this.execute(ctx) })
        )
    }

    private fun execute(ctx: CommandContext<CommandSourceStack>): Int {
        if (ctx.getSource().getPlayer() != null) {
            val player: ServerPlayer = ctx.getSource().getPlayer()!!
            var playerPc: PCStore? = null
            try {
                playerPc = Cobblemon.storage.getPC(player)
            } catch (e: NoPokemonStoreException) {
                player.sendSystemMessage(Component.literal("Error accessing PC..."))
                return -1
            }
            val boxNum: Int = ctx.getArgument<Int>("box", Int::class.java) - 1
            val box = playerPc.boxes[boxNum]
            if (box == null) {
                player.sendSystemMessage(Component.literal("Error accessing box... $boxNum"))
                return -1
            }
            for (i in 0..29) {
                playerPc.remove(PCPosition(boxNum, i))
            }
            player.sendSystemMessage(Component.literal("Box " + (boxNum + 1) + " has been emptied."))
        } else {
            ctx.getSource().sendFailure(Component.literal("Sorry, this is only for players."))
        }
        return 1
    }
}
